interface/billing/sl_eob_invoice.php

```diff
@@ -89,156 +89,182 @@ function row_delete($table, $where)
-            if (pfxlen < 0) continue;
-            let pfx = ename.substring(0, pfxlen);
-            let code = pfx.substring(pfx.indexOf('[') + 1, pfxlen - 1);
-            let cPay = parseFloat(f[pfx + '[pay]'].value).toFixed(2);
-            let cAdjust = parseFloat(f[pfx + '[adj]'].value).toFixed(2);
-
-            if ((cPay != 0) || cAdjust != 0) {
-                allempty = false;
-            }
-            if(adjDisable) {
-                if ((cAdjust == 0 && ins_done.value == 'changed'))) {
+            return false;
+        }
+
+        // Onsubmit handler.  A good excuse to write some JavaScript.
+        function validate(f) {
+            let delcount = 0;
+            let allempty = true;
+
+            for (let i = 0; i < f.elements.length; ++i) {
+                let ename = f.elements[i].name;
+                // Count deletes.
+                if (ename.substring(0, 9) == 'form_del[') {
+                    if (f.elements[i].checked) {
+                        ++delcount;
+                    }
+                    continue;
+                }
+                let pfxlen = ename.indexOf('[pay]');
+                if (pfxlen < 0) {
+                    continue
+                };
+                let pfx = ename.substring(0, pfxlen);
+                let code = pfx.substring(pfx.indexOf('[') + 1, pfxlen - 1);
+                let cPay = parseFloat(f[pfx + '[pay]'].value).toFixed(2);
+                let cAdjust = parseFloat(f[pfx + '[adj]'].value).toFixed(2);
+
+                if ((cPay !== 0) || cAdjust !== 0) {
                     allempty = false;
                 }
+                if(adjDisable) {
+                    if ((cAdjust == 0 && ins_done.value == 'changed')) {
+                        allempty = false;
+                    }
+                }
+                if ((cPay !== 0) && isNaN(parseFloat(f[pfx + '[pay]'].value))) {
+                    dlgopen('', '', 675, 250, '', '<div class="text-danger">Warning</div>', {
+                        type: 'Alert',
+                        html: '<p>Payment value for code ' + code + ' is not a number</p>',
+                    });
+                    return false;
+                }
+                if ((cAdjust !== 0) && isNaN(parseFloat(f[pfx + '[adj]'].value))) {
+                    alert('OK')
+                    dlgopen('', '', 675, 250, '', '<div class="text-danger">Warning</div>', {
+                        type: 'Alert',
+                        html: '<p>Adjustment value for code ' + code + ' is not a number</p>',
+                    });
+                    return false;
+                }
+                if ((cAdjust !== 0) && !f[pfx + '[reason]'].value) {
+                    dlgopen('', '', 675, 250, '', '<div class="text-danger">Warning</div>', {
+                        type: 'Alert',
+                        html: '<p>Please select an adjustment reason for code ' + code + '</p>',
+                    });
+                    return false;
```

Comment:
This should be more along the lines of
if ((cAdjust !== 0) && !f[pfx + '[reason]'].value && ! adjDisable) {
    let message = "<p>Please select an adjustment reason for code" + code + '</p>";
    alertMsg(message, 1000);
   return false;
}

This is a timed BS alert so user doesn't have to interact if they don't want.
umm i'm rather proud of this function along with if you wanted to run promised BS alerts for the entire validation function you could use
(async (time) => {
       await asyncAlertMsg('Waiting till x'ed out or timeout!', time); ...now go;
})(3000).then(rtn => { ... but then could be more });

Reply:
> if ((cAdjust !== 0) && !f[pfx + '[reason]'].value && ! adjDisable) { let message = "<p>Please select an adjustment reason for code" + code + '</p>"; alertMsg(message, 1000); return false; }

anyone know how to get code segs to format here?

Reply:
Hi @sjpadgett 
Did you recommend using alertMsg to instead dlgopen?

Reply:
yes.